documentation/optee_design.md

```diff
@@ -665,16 +665,51 @@ instead. When the page is mapped new data is copied from backing store and
 the hash of the page is verified. If it's OK the pager returns from the
 exception to resume the execution.
 
-# 9. Cryptographic abstraction layer
+## Paging of user TA
+
+Paging of user TAs can optionally be enabled with CFG_PAGED_USER_TA=y.
+Paging of user TAs is analogous to paging of OP-TEE kernel parts but with a
+few differences:
+- Read/write pages are paged in addition to read-only pages
+- Page tables are managed dynamically
+
+tee_pager_add_uta_area() is used to setup initial read/write mapping needed
+when populating the TA. When the TA is fully populated and relocated
+tee_pager_set_uta_area_attr() changes the mapping of the area to strict
+permissions used when the TA is running.
+
+# 9. Memory objects
+
+A memory object, MOBJ, describes a piece of memory. The interface provided
+is mostly abstract when it comes to using the MOBJ to populate translation
+tables etc.
+
+There's different kinds of MOBJs describing:
+- physically contiguous memory
+  - created with mobj_phys_alloc()
+- virtual memory
+  - one instance with the name mobj_virt available
+  - spans the entire virtual address space
+- physically contiguous memory allocated from a tee_mm_pool_t *
+  - created with mobj_mm_alloc()
+- paged memory
+  - created with mobj_paged_alloc()
+  - only contains the supplied size and makes mobj_is_paged() return true if
+    supplied as argument
+- secure paged memory
+  - created with mobj_sec_shm_alloc()
```

Comment:
It's used for secure shared memory, but now in hindsight it doesn't look good I should try to come up with a better name, suggestions are welcome.
Yes, I'll add `mobj_is_secure()` also.